packages/foam-vscode/src/utils.ts

```diff
@@ -184,3 +188,55 @@ export async function focusNote(notePath: string, moveCursorToEnd: boolean) {
     editor.selection = new Selection(range.end, range.end);
   }
 }
+
+export function getContainsTooltip(notes: Note[]): string {
+  const titles = notes.map(n => n.title).slice(0, 5);
+  return `Contains "${titles.join('", "')}"`;
+}
+
+/**
+ * Depending on the current vscode version, returns a MarkdownString of the
+ * note content casted as string or returns a simple string
+ * MarkdownString is only available from 1.52.1 onwards
+ * https://code.visualstudio.com/updates/v1_52#_markdown-tree-tooltip-api
+ * @param note A Foam Note
+ */
+export function getNoteTooltip(note: Note): string {
+  const STABLE_MARKDOWN_STRING_API_VERSION = '1.52.1';
+
+  if (version >= STABLE_MARKDOWN_STRING_API_VERSION) {
+    return formatMarkdownTooltip(note) as any;
+  }
+
+  return formatSimpleTooltip(note);
+}
+
+export function formatMarkdownTooltip(note: Note): MarkdownString {
+  const LINES_LIMIT = 15;
+  const markdown = firstNLines(stripFrontMatter(note.source.text), LINES_LIMIT);
```

Comment:
what happens if there is an image? and if it's a tall image?

Reply:
what happens if there are 15 very long lines?

It might not be easy to "ensure" the tooltip is under control (I guess we can't limit its height as a container), but as a heuristic lines, characters, and images should be enough

Reply:
Images are not shown. For now there's an ugly missing image icon instead so I was thinking of replacing the images by their alt text instead. So:

```
![This is an alt text](some/kind/of/uri.png)
```
could be replaced by:
```
Image: "This is an alt text"
```

And completely removed if no alt text is provided. wdyt?

Reply:
But for sure the excerpt func has to be refined, any inputs are welcomed

Reply:
Here are my thoughts:
- re image, I believe it's fine to leave the original text, it's understandable by the user what's going on
- re size, the best pareto effort I can think of is either what you have, or then also limit each line to 100 char, or in general pick the first 1500 chars of the string and do the line limiting on those. It's a bit hard to judge without seeing the actual results, each of them have pros and cons. wdyt?

Reply:
I improved the excerpt function, should be pretty good now!